### PR TITLE
Core/Player: implement Duel Reset HP and Mana config

### PR DESCRIPTION
**Pull request by GigaDev90**
This PR introduces a new configurable setting (disabled by default) which reset the mana and health of the player before and after Duel.
Thanks to shindarth for help me during the work.

### CODE REVIEW

File: src/server/game/World/World.cpp [hunk 1]
```diff
@@ -1191,6 +1191,7 @@ void World::LoadConfigSettings(bool reload)
     m_int_configs[CONFIG_MAX_WHO] = sConfigMgr->GetIntDefault("MaxWhoListReturns", 49);
     m_bool_configs[CONFIG_START_ALL_SPELLS] = sConfigMgr->GetBoolDefault("PlayerStart.AllSpells", false);
     m_int_configs[CONFIG_HONOR_AFTER_DUEL] = sConfigMgr->GetIntDefault("HonorPointsAfterDuel", 0);
+    m_bool_configs[CONFIG_RESTORE_HEALTHMANA_AFTER_DUEL] = sConfigMgr->GetBoolDefault("RestoreHealthManaAfterDuel", 0);
     m_bool_configs[CONFIG_RESET_DUEL_COOLDOWNS] = sConfigMgr->GetBoolDefault("ResetDuelCooldowns", 0);
```
Comment (ShinDarth) [inline src/server/game/World/World.cpp]: Can you replace:
```
m_bool_configs[CONFIG_RESTORE_HEALTHMANA_AFTER_DUEL] = sConfigMgr->GetBoolDefault("RestoreHealthManaAfterDuel", 0);
m_bool_configs[CONFIG_RESET_DUEL_COOLDOWNS] = sConfigMgr->GetBoolDefault("ResetDuelCooldowns", 0);
```
with:

```
m_bool_configs[CONFIG_RESTORE_HEALTHMANA_AFTER_DUEL] = sConfigMgr->GetBoolDefault("RestoreHealthManaAfterDuel", false);
m_bool_configs[CONFIG_RESET_DUEL_COOLDOWNS] = sConfigMgr->GetBoolDefault("ResetDuelCooldowns", false);
```

**Comment by MitchesD** [comment]
To be honest, I don't like putting these custom things in Player class, this file is too big and for non-needed things we should make some separated place

**Comment by ShinDarth** [comment]
@MitchesD if you explain us how we will be glad to edit this

**Comment by Naios** [comment]
We could start to use PlayerScripts...

**Comment by ShinDarth** [comment]
Can you show to us how to convert this to Player? We have no idea and I see no documentation

**Comment by Warpten** [comment]
Here is your documentation

https://github.com/TrinityCore/TrinityCore/blob/3bc3b67a8fae059e1019446b09882dfa7fcd11a0/src/server/scripts/World/action_ip_logger.cpp#L161
https://github.com/TrinityCore/TrinityCore/blob/3bc3b67a8fae059e1019446b09882dfa7fcd11a0/src/server/scripts/World/chat_log.cpp

https://github.com/TrinityCore/TrinityCore/blob/3bc3b67a8fae059e1019446b09882dfa7fcd11a0/src/server/game/Scripting/ScriptMgr.h#L679

sigh

**Comment by ShinDarth** [comment]
@Warpten I wouldn't call it "documentation"... however

I'm trying to understand how it works, by doing this: https://github.com/TrinityCore/TrinityCore/pull/15731

but I need help, can you check it please?

**Comment by ShinDarth** [comment]
https://github.com/TrinityCore/TrinityCore/pull/15732

**Pull request closed without merging**